askama_derive/src/generator.rs

```diff
@@ -1053,6 +1074,33 @@ impl<'a> Generator<'a> {
         Ok(size_hint)
     }
 
+    fn handle_embed(
+        &mut self,
+        buf: &mut Buffer,
+        embed: &'a Embed<'_>,
+    ) -> Result<usize, CompileError> {
+        self.flush_ws(embed.ws1);
+        self.write_buf_writable(buf)?;
+        let embed_path = self
+            .input
+            .config
+            .find_template(embed.path, Some(&self.input.path))?;
+        let mut embedded_context =
+            Context::new(self.input.config, &self.input.path, embed.nodes.as_slice())?;
+        embedded_context.extends = Some(embed_path);
+        let heritage = Heritage::new(&embedded_context, self.contexts);
+
+        let mut generator = Generator::new(
+            self.input,
+            self.contexts,
+            Some(&heritage),
+            MapChain::new(),
+            self.whitespace,
+        );
+        self.prepare_ws(embed.ws2);
+        generator.handle(heritage.root, heritage.root.nodes, buf, AstLevel::Top)
```

Comment:
Basically this would become valid
```django
{{ my_context_value }}
{% let x = 10 %}

{% embed "embed.html" %}
      {% block content %}
            {{ my_context_value }}
            {{ x }}
      {% endblock %}
{% endembed %}
```

Reply:
This sounds very cool. I did not intend to drop the parent context here, so I will apply this to the code!